### PR TITLE
Add Support for Legacy and New Scopes

### PR DESCRIPTION
**Pull request by dshanske**
Right now the plugin completely ignores scopes. This gives it support for current and future scopes.

### CODE REVIEW

**Comment by snarfed** [comment]
this is really great, i'm glad we're adding scope support. out of curiosity though, why add a filter? what's the use case for other plugins or themes overriding scope auth checks?

**Comment by dshanske** [comment]
New scopes.

**Comment by dshanske** [comment]
@snarfed Rebased this based on the other merges. I'm willing to remove the filter, but I think it would help optional expansion. Will defer.

**Comment by snarfed** [comment]
yeah, let's drop the filter for now. i'd need to hear at least a couple different concrete use cases for extensible scope authorization before it would make sense to me personally, and maybe not even then. auth is important and delicate, and not generally something you want to let any other code, written by anyone, arbitrarily override.

**Comment by aaronpk** [comment]
Agreed on removing the filter. Until there is a good use case for a plugin modifying the behavior of the plugin that also wants to provide finer-grained access, there's no need to make that part extensible. Supporting the primary list of scopes create/update/delete/undelete is good for now, and you could also accept "post" if it's requested for legacy clients.

**Comment by dshanske** [comment]
@snarfed You suggested that I removed post scope. I didn't, I just moved the check further down into the capability function. It treats post as full permission, and create, update, delete, or undelete as the scopes for the individual functions... although with only a few clients supporting undelete, maybe delete should be the scope for either.

**Comment by snarfed** [comment]
true! you're absolutely right, sorry about that. thanks!

**Review by snarfed — APPROVED** [review]
lgtm! just one refactoring suggestion. feel free to merge after that!

**Comment by dshanske** [comment]
I am thinking that I might make some other changes now that I am not using a filter. Move the WordPress permission check into the check scope function. May ask for your thoughts once done.

File: micropub.php [hunk 1]
```diff
@@ -246,6 +250,19 @@ private static function authorize() {
...
 
+	/**
+	 * Check scope
+	 *
+	 * @param string $scope
+	 *
+	 * @return boolean
+	**/
+	protected static function check_scope( $scope ) {
+		if ( in_array( $scope, static::$scope, true ) || in_array( 'post', static::$scope, true ) ) {
```
Comment (snarfed) [inline micropub.php]: very minor nit: this can now be just `return in_array(...) || in_array(...);`

File: readme.md [hunk 2]
```diff
@@ -17,6 +17,18 @@ Supports the [full W3C Micropub CR spec](https://www.w3.org/TR/micropub/) as of
 This project is placed in the public domain. You may also use it under the [CC0 license](http://creativecommons.org/publicdomain/zero/1.0/).
 
 
+## Scope 
+
+Supports the following [scope](https://indieweb.org/scope) parameters requested by Micropub clients.
+* Post (legacy) - Grants all user delegated access
```
Comment (snarfed) [inline readme.md]: these should probably be lower case, eg `post`, since they're case sensitive.

File: readme.txt [hunk 3]
```diff
@@ -25,6 +25,17 @@ Supports the [full W3C Micropub CR spec](https://www.w3.org/TR/micropub/) as of
 
 This project is placed in the public domain. You may also use it under the [CC0 license](http://creativecommons.org/publicdomain/zero/1.0/).
 
+== Scope ==
+
+Supports the following [scope](https://indieweb.org/scope) parameters requested by Micropub clients.
```
Comment (snarfed) [inline readme.txt]: ditto

**Review by snarfed — COMMENTED** [review]
looks good! only a couple minor nits.

btw, github evidently doesn't notify a PR's reviewers etc when you just amend an existing commit like this. it also makes reviewing a bit harder, since it's hard to tell just what's changed since you last reviewed. might be one reason to wait on squashing until after reviewing. just fyi.

**Pull request merged**